substrate/frame/parameters/src/lib.rs

```diff
@@ -167,6 +168,23 @@ pub mod pallet {
 		type WeightInfo: WeightInfo;
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			let proof_size_before: u64 = get_proof_size().unwrap_or(0);
+
+			let items = Parameters::<T>::iter().count() as u64;
```

Comment:
I still do not agree with the need for a `on_initialize` hook here. Also `Parameters::<T>::iter()` will be empty until `set_parameter` is called.

Reply:
Not all parameter usage is captured by our benchmarks. For instance, some parameters may be read within xcm or in other unaccounted areas. As a result, we can't rely solely on benchmark coverage to determine parameter impact.
To mitigate potential issues, we should assume that all parameters stored in each block will be read. 

This approach helps us avoid hitting the proof size limit and prevents our chain from stalling, which would otherwise require recovery through relay chain governance.

Reply:
`Parameters::<T>::iter()` initially will be empty, but as we add more and more parameters it will report all the stored ones